### PR TITLE
Clarify when and why LearningRateScheduler is called

### PR DESCRIPTION
**Pull request by lukasugar**
## Clarify when and why LearningRateScheduler is called

Discussed here: https://pytorch-lightning.slack.com/archives/CRBLFHY79/p1654788931469689

Fixes #\<issue_number>

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: docs/source/model/manual_optimization.rst [hunk 1]
```diff
@@ -186,8 +186,8 @@ Use ``self.lr_schedulers()`` in  your :class:`~pytorch_lightning.core.module.Lig
 defined in your :meth:`~pytorch_lightning.core.module.LightningModule.configure_optimizers`.
 
 .. warning::
-   * Before v1.3, Lightning automatically called ``lr_scheduler.step()`` in both automatic and manual optimization. From
-     1.3, ``lr_scheduler.step()`` is now for the user to call at arbitrary intervals.
+   * Before v1.3, Lightning automatically called ``lr_scheduler.step()`` in both automatic and manual optimization. 1.3, ``lr_scheduler.step()`` can now be called at arbitrary intervals by the user, or automatically by Lightning if ``"interval"`` is defined in ``configure_optimizers()``.
```
Comment (carmocca) [inline docs/source/model/manual_optimization.rst:189]: This is not correct during manual optimization, we actually show a warning:

```python
/home/carmocca/git/py39/lib/python3.9/site-packages/pytorch_lightning/core/optimizer.py:327: RuntimeWarning: The lr scheduler dict contains the key(s) ['interval'], but the keys will be ignored. You need to call `lr_scheduler.step()` manually in manual optimization.
```

**Review by carmocca — CHANGES_REQUESTED** [review]

File: docs/source/model/manual_optimization.rst (hunk 1 above)
Comment (akihironitta) [reply docs/source/model/manual_optimization.rst]: Hi @lukasugar, the section pointed out by carmocca is about manual optimization, so the following statement is always false in manual optimization. (while it's true in automatic optimization.)

> automatically by Lightning if ``"interval"`` is defined in ``configure_optimizers()``.

File: docs/source-pytorch/model/manual_optimization.rst [hunk 2]
```diff
@@ -186,8 +186,8 @@ Use ``self.lr_schedulers()`` in  your :class:`~pytorch_lightning.core.module.Lig
 defined in your :meth:`~pytorch_lightning.core.module.LightningModule.configure_optimizers`.
 
 .. warning::
-   * Before v1.3, Lightning automatically called ``lr_scheduler.step()`` in both automatic and manual optimization. From
-     1.3, ``lr_scheduler.step()`` is now for the user to call at arbitrary intervals.
+   * Before v1.3, Lightning automatically called ``lr_scheduler.step()`` in both automatic and manual optimization. 1.3, ``lr_scheduler.step()`` can now be called at arbitrary intervals by the user, or automatically by Lightning if ``"interval"`` is defined in ``configure_optimizers()``.
+     Check ``configure_optimizers()`` for more details.
```
Comment (rohitgr7) [inline docs/source-pytorch/model/manual_optimization.rst:190]:
```suggestion
   * Before v1.3, Lightning automatically called ``lr_scheduler.step()`` in both automatic and manual optimization. Post v1.3, ``lr_scheduler.step()`` can now be called at arbitrary intervals by the user in case of manual optimization, or by Lightning if ``"interval"`` is defined in :meth:`~pytorch_lightning.core.module.LightningModule.configure_optimizers` in case of automatic optimization.
```
I think this is better.

File: src/pytorch_lightning/core/module.py [hunk 3]
```diff
@@ -1329,7 +1329,9 @@ def configure_optimizers(self):
         Note:
             Some things to know:
 
-            - Lightning calls ``.backward()`` and ``.step()`` on each optimizer and learning rate scheduler as needed.
+            - Lightning calls ``.backward()`` and ``.step()`` on each optimizer as needed.
+            - If learning rate scheduler is specified in ``configure_optimizers()`` with parameter ``"interval"``,
+              Lightning will call scheduler's ``.step()`` method automatically.
```
Comment (rohitgr7) [inline src/pytorch_lightning/core/module.py:1334]:
```suggestion
            - If learning rate scheduler is specified in ``configure_optimizers()`` with key ``"interval"`` (default="epoch") in the scheduler configuration,
              Lightning will call the scheduler's ``.step()`` method automatically in case of automatic optimization.
```
how about?

**Review by rohitgr7 — APPROVED** [review]

File: docs/source-pytorch/model/manual_optimization.rst [hunk 4]
```diff
@@ -186,8 +186,7 @@ Use ``self.lr_schedulers()`` in  your :class:`~pytorch_lightning.core.module.Lig
 defined in your :meth:`~pytorch_lightning.core.module.LightningModule.configure_optimizers`.
 
 .. warning::
-   * Before v1.3, Lightning automatically called ``lr_scheduler.step()`` in both automatic and manual optimization. From
-     1.3, ``lr_scheduler.step()`` is now for the user to call at arbitrary intervals.
+   * Before v1.3, Lightning automatically called ``lr_scheduler.step()`` in both automatic and manual optimization. Post v1.3, ``lr_scheduler.step()`` can now be called at arbitrary intervals by the user in case of manual optimization, or by Lightning if ``"interval"`` is defined in :meth:`~pytorch_lightning.core.module.LightningModule.configure_optimizers` in case of automatic optimization.
```
Comment (carmocca) [inline docs/source-pytorch/model/manual_optimization.rst:189]: Removing the statement for the old version.

```suggestion
   * ``lr_scheduler.step()`` can be called at arbitrary intervals by the user in case of manual optimization, or by Lightning if ``"interval"`` is defined in :meth:`~pytorch_lightning.core.module.LightningModule.configure_optimizers` in case of automatic optimization.
```

File: src/pytorch_lightning/core/module.py [hunk 5]
```diff
@@ -1329,7 +1329,9 @@ def configure_optimizers(self):
         Note:
             Some things to know:
 
-            - Lightning calls ``.backward()`` and ``.step()`` on each optimizer and learning rate scheduler as needed.
+            - Lightning calls ``.backward()`` and ``.step()`` on each optimizer as needed.
+            - If learning rate scheduler is specified in ``configure_optimizers()`` with key ``"interval"`` (default="epoch") in the scheduler configuration,
```
Comment (carmocca) [inline src/pytorch_lightning/core/module.py:1333]:
```suggestion
            - If learning rate scheduler is specified in ``configure_optimizers()`` with key ``"interval"`` (default "epoch") in the scheduler configuration,
```

**Review by carmocca — APPROVED** [review]

**Comment by lukasugar** [comment]
I agree with your suggestions, it does make it clear!
